generic/src/main/scala-3/neotypes/generic/deriveProductMap.scala

```diff
@@ -0,0 +1,41 @@
+package neotypes
+package generic
+
+import neotypes.mappers.ResultMapper
+import neotypes.mappers.ResultMapper.DerivedProductMap
+import neotypes.model.types.NeoObject
+import neotypes.model.exceptions.ResultMapperException
+import neotypes.model.exceptions.IncoercibleException
+
+import shapeless3.deriving.*
+
+trait CaseClassDerivedProductMap[P <: Product] extends DerivedProductMap[P]
+object CaseClassDerivedProductMap:
+  private type Acc = (Int, NeoObject, Option[ResultMapperException])
+  given [P <: Product](using
+    labelled: Labelling[P],
+    inst: K0.ProductInstances[ResultMapper, P]
+  ): CaseClassDerivedProductMap[P] with
+    override def map(obj: NeoObject): Either[ResultMapperException, P] =
+      // `unfold` traverses over Product field types.
+      // For example, `inst.unfold` for `case class A(i:Int, s: String)`
+      // traverses over `Int` and `String`
+      val ((_, _, decodeOpt), maybeDerived) = inst.unfold[Acc]((0, obj, None)):
```

Comment:
It seems you don't need to thread `obj` on the `unfold`

Reply:
Yeah, right.